samcli/lib/bootstrap/bootstrap.py

```diff
@@ -0,0 +1,108 @@
+"""
+Bootstrap's user's development environment by creating cloud resources required by SAM CLI
+"""
+
+import logging
+import boto3
+
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from samcli.commands.exceptions import UserException
+
+
+LOG = logging.getLogger(__name__)
+SAM_CLI_STACK_NAME = "aws-sam-cli-managed-stack"
+
+MANAGED_STACK_DEFINITION = """
```

Comment:
* this lives in a separate file?
* can we add the version number of sam cli with which this stack was created?
Any other cheap metadata we can add today?

Reply:
* separate file: Yeah, the template is small. it's cheap to have it here. we will move out later
* metadata: Good idea!